component/common/config/types.go

```diff
@@ -0,0 +1,153 @@
+// Package config contains types from github.com/prometheus/common/config, but changes them to be compatible with river
```

Comment:
general nit: please use proper capitalization for nouns like River/Prometheus, and use proper grammar in comments so comments start with capital letters and end in full stops. You should be writing comments like documentation that external users are going to read, because they will.